src/green/builder.rs

```diff
@@ -7,8 +7,8 @@ use crate::{
 
 #[derive(Default, Debug)]
 pub struct NodeCache {
-    nodes: FxHashSet<GreenNode>,
-    tokens: FxHashSet<GreenToken>,
+    nodes: HashMap<GreenNode, ()>,
+    tokens: HashMap<GreenToken, ()>,
```

Comment:
Are these `HashMap` just because of `drain_filter`?

Reply:
Yes (unfortunately).

A further optimization step in the pipeline will take further advantage of `HashMap::raw_entry_mut`, so the switch to using set-like maps is probably inevitable, though.